addons/hr/static/src/models/public_employee/public_employee.js

```diff
@@ -0,0 +1,205 @@
+            });
+        }
+
+        /**
+         * Gets the canonical chat between the user of this employee and the
+         * current user.
+         *
+         * If a chat is not appropriate a notification is displayed instead.
+         *
+         * @returns {mail.thread|undefined}
+         */
+        async getCanonicalChat() {
+            if (!this.user && !this.hasCheckedUser) {
+                await this.async(() => this.checkIsUser());
+            }
+            // prevent chatting with non-users
+            if (!this.user) {
+                this.env.services['notification'].notify({
+                    message: this.env._t("You can only chat with employees that have a dedicated user."),
+                    type: 'info',
+                });
+                return;
+            }
+            return this.user.getCanonicalChat();
+        }
+
+        /**
+         * Opens a chat between the user of this employee and the current user
+         * and returns it.
+         *
+         * If a chat is not appropriate a notification is displayed instead.
+         *
+         * @param {Object} [options] forwarded to @see `mail.thread:open()`
+         * @returns {mail.thread|undefined}
+         */
+        async openChat(options) {
+            const chat = await this.async(() => this.getCanonicalChat());
+            if (!chat) {
+                return;
+            }
+            await this.async(() => chat.open(options));
+            return chat;
+        }
+
+        /**
+         * Opens the most appropriate view that is a profile for this employee.
+         */
+        async openProfile() {
+            return this.env.messaging.openDocument({
+                id: this.id,
+                model: 'hr.employee.public',
+            });
+        }
+
+        //----------------------------------------------------------------------
+        // Private
+        //----------------------------------------------------------------------
+
+        /**
+         * @override
+         */
+        static _createRecordLocalId(data) {
+            return `${this.modelName}_${data.id}`;
+        }
+
+    }
+
+    Employee.fields = {
+        /**
+         * Whether an attempt was already made to fetch the user corresponding
+         * to this employee. This prevents doing the same RPC multiple times.
+         */
+        hasCheckedUser: attr({
+            default: false,
+        }),
+        /**
+         * Unique identifier for this employee.
+         */
+        id: attr(),
+        /**
+         * Partner related to this employee through its user.
+         */
+        userPartner: one2one('mail.partner', {
+            inverse: 'employee',
+            related: 'user.partner',
+        }),
+        /**
+         * User related to this employee.
+         */
+        user: one2one('mail.user', {
+            inverse: 'employee',
+        }),
+    };
+
+    Employee.modelName = 'hr.employee';
```

Comment:
model has been renamed, but file still named `public_employee`. Should be just `employee`.

Reply:
ok